### DISCUSSION TITLE
EdgeInsets Markup Extension

### DISCUSSION THREAD

**Discussion by tuyen-vuduc** [CommunityToolkit/Maui#168] [New Feature Discussions]
Hi guys,

I did create and use `EdgetInsetsMarkupExtension` for my project. I think you might also like it.

# What it can do

This extension helps us to define our margin/padding in a clean, meaning way. The idea comes from [Flutter's EdgeInsets](https://api.flutter.dev/flutter/painting/EdgeInsets-class.html).

I already created a PR to XamarinCommunityToolkit library as you can find [here](https://api.flutter.dev/flutter/painting/EdgeInsets-class.html).

I used it in [my sample TodoApp project](https://github.com/tuyen-vuduc/todo-app-xamarin-forms) as well.

# How to use it

1) 
```
<View Maggin="16,8" />
```
==>
```
<View Maggin="{app:EdgetInsets Horizontal=16, Vertical=8}" />
```

2) 
```
<View Maggin="16,20,12,40" />
```
==>
```
<View Maggin="{app:EdgetInsets Left=16, Top=20, Right=12, Bottom=40}" />
```

3) 
```
<View Maggin="16" />
```
==>
```
<View Maggin="{app:EdgetInsets All=16}" />
```

We can create more combinations from properties provided. The priority will be
- Specified edge: `Top`, `Left`, `Right`, `Bottom`
- Specified direction: `Horizontal`, `Vertical`
- Default: `All`

In case no value is specified, Zero will be assigned to a particular edge.

# Why it benefits you?
- It's clean because we really know what a margin, padding we are going to define
- It's meaningful because it's very obvious when reading the code (XAML)
- It's helpful because you could easily define your common spacing and use `x:Static` or `x:StaticResource` to reference to it

Regarding the last point, normally, I define these kinds of spacing: large, medium, small, xlarge, xsmall.

If we go the old way, and we want to reuse those values, we have to define a lot of combinations because we couldn't define 
```
<View Margin="large, small" />
```
However, with my extensions, we don't have to define all combinations, we just use the value when it's really should be.
```
<View Margin={app:EdgeInsets Top={x:Static app:SpacingLarge}, Bottom={x:Static app:SpacingSmall}} />
```

Please share your thoughts if I should create a PR to this repo.

Thank you.

**Comment by CliffAgius** [comment]
Closed as per the Community Stand-up discussion here: https://youtu.be/N9wMcBP4jtg?t=2889
